docs/en/02_Developer_Guides/09_Security/04_Secure_Coding.md

```diff
@@ -585,6 +583,75 @@ In a future release this behaviour will be changed to be on by default, and this
+		Director::forceSSL();
+	}
+
+Forcing HTTPS so requires a certificate to be purchased or obtained through a vendor such as 
+[lets encrypt](https://letsencrypt.org/) and configured on your web server.
+
+In addition to this ensure the users are provided with secure cookies by setting the recommended "Secure" and 
+"HTTPOnly" flags. These flags prevent them from being stolen by an attacker through javascript. 
+
+ - The "Secure" cookie flag instructs the browser not to send the cookie over an insecure HTTP connection. If this 
+flag is not present, the browser will send the cookie even if HTTPS is not in use, which means it is transmitted in 
+clear text and can be intercepted and stolen by an attacker who is listening on the network.
+
+- The "HTTPOnly" flag lets the browser know whether or not a cookie should be accessible by client-side JavaScript 
+code. It is best practice to set this flag unless the application is known to use JavaScript to access these cookies 
+as this prevents an attacker who achieves cross-site scripting from accessing these cookies.
+
+
+	:::php
+	
+	Cookie::set('cookie-name', 'chocolate-chip', $expiry = 30, $path = null, $domain = null, $secure = true, 
+		$httpOnly = false
+	);
+
+## Security Headers
+
+In addition to forcing HTTPS browsers can support additional security headers which can only allow access to a website 
+via a secure connection. As browsers increasingly provide negative feedback regarding unencrypted HTTP connections, 
+ensuring an HTTPS connection will provide a better and more secure user experience.  
+
+- The `Strict-Transport-Security` header instructs the browser to record that the website MUST be visited via a 
+secure connection. This prevents websites from becoming insecure in the future from 
+stray absolute links or references without https from external sites. 
+
+For sensitive pages, such as members areas, or places where sensitive information is present, adding cache control
+ headers can explicitly instruct browsers not to keep a local cached copy of content and can prevent content from 
+ being cached throughout the infrastructure (e.g. Proxy, caching layers, WAF etc). 
+ 
+- The headers `Cache-control: no-store` and `Pragma: no-cache` along with expiry headers of `Expires: <current date>` 
+and `Date: <current date>` will ensure that sensitive content is not stored locally or able to be retrieved by 
+unauthorised local persons.
+
+
+	:::php
+
+	class MySecureController extends Controller {
+		
+		public function init() {
+			parent::init();
+        
+			// max-age can be configured to anything in seconds: max-age=31536000 (1 year), 
+			// for rollout, consider something lower
+			// includeSubDomains to ensure all present and future sub domains will also be HTTPS
+			$this->response->addHeader('Strict-Transport-Security', 'max-age=86400; includeSubDomains');
+			$this->response->addHeader('Cache-control', 'max-age=0, must-revalidate, no-transform, no-cache, no-store');
```

Comment:
@tractorcow Why are we suggesting to use static method calls instead of setting values on a Response object? I would have thought the static method calls is the outdated way of doing things, given the relative difficulty of testing those in isolation?